src/Parsers/CommonParsers.cpp

```diff
@@ -65,4 +65,17 @@ bool ParserKeyword::parseImpl(Pos & pos, ASTPtr & /*node*/, Expected & expected)
     return true;
 }
 
+bool ParserDoubleColon::parseImpl(Pos & pos, ASTPtr & /*node*/, Expected & expected)
+{
+    /// Do not move position if only one colon mathced.
+    Pos begin = pos;
+    if (parser_colon.ignore(begin, expected) && parser_colon.ignore(begin, expected))
```

Comment:
It will be better to introduce double colon as a separate token in Lexer.

Similarly, `-` and `--` are separate tokens in Lexer.